core/system_paths.py

```diff
@@ -28,32 +29,146 @@ def on_ready():
 
     default_system_paths = {
         "user": home,
-        "desktop": home / "Desktop",
-        "desk": home / "Desktop",
-        "documents": home / "Documents",
-        "docks": home / "Documents",
-        "downloads": home / "Downloads",
-        "music": home / "Music",
-        "pictures": home / "Pictures",
-        "videos": home / "Videos",
         "talon home": talon_home,
         "talon recordings": talon_home / "recordings",
         "talon user": actions.path.talon_user(),
     }
 
     if app.platform == "windows":
-        default_system_paths["profile"] = home
-        onedrive_path = home / "OneDrive"
-
-        # this is probably not the correct way to check for OneDrive, quick and dirty
-        if (onedrive_path / "Desktop").is_dir():
-            default_system_paths["desktop"] = onedrive_path / "Desktop"
-            default_system_paths["documents"] = onedrive_path / "Documents"
-            default_system_paths["one drive"] = onedrive_path
-            default_system_paths["pictures"] = onedrive_path / "Pictures"
+        import winreg
+
+        # Virtual folders.
+        default_system_paths.update(
+            {
+                # Virtual-only folders.
+                "quick access": "shell:::{679F85CB-0220-4080-B29B-5540CC05AAB6}",
+                "this pc": "shell:MyComputerFolder",  # Or `shell:::{20D04FE0-3AEA-1069-A2D8-08002B30309D}`, also without `shell:`.
+                "network": "shell:NetworkPlacesFolder",  # Or `shell:::{F02C1A0D-BE21-4350-88B0-7367FC96EF3C}`, also without `shell:`.
+                "recycle bin": "shell:RecycleBinFolder",  # Or `shell:::{645FF040-5081-101B-9F08-00AA002F954E}`, also without `shell:`.
+                # Folders with virtual and resolved form.
+                "virtual desktop": "shell:Desktop",
+                # More like, e.g., `shell:Downloads` can be found under the registry key
+                # `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions`.
+                # You can also find a long list of GUIDs under
+                # <https://www.autohotkey.com/docs/v2/misc/CLSID-List.htm>.
+            }
+        )
+
+        # Resolved forms of virtual folders whose locations the user may have changed.
+        with winreg.OpenKey(
+            winreg.HKEY_CURRENT_USER,
+            r"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\User Shell Folders",
+        ) as key:
+
+            def get_path(value_name, fallback):
+                try:
+                    path_str, value_type = winreg.QueryValueEx(key, value_name)
+                    assert value_type == winreg.REG_EXPAND_SZ
+                except (FileNotFoundError, AssertionError):
+                    path_str = str(fallback)
+                return Path(os.path.expandvars(path_str))
```

Comment:
No need to do this since Python 3.6; `os.path.expandvars` will accept a path-like object.

```suggestion
                    path, value_type = winreg.QueryValueEx(key, value_name)
                    assert value_type == winreg.REG_EXPAND_SZ
                except (FileNotFoundError, AssertionError):
                    path = fallback
                return Path(os.path.expandvars(path))
```